azure/services/virtualmachines/spec.go

```diff
@@ -77,8 +77,8 @@ func (s *VMSpec) OwnerResourceName() string {
 // Parameters returns the parameters for the virtual machine.
 func (s *VMSpec) Parameters(ctx context.Context, existing interface{}) (params interface{}, err error) {
 	if existing != nil {
-		if _, ok := existing.(compute.VirtualMachine); !ok {
-			return nil, errors.Errorf("%T is not a compute.VirtualMachine", existing)
+		if _, ok := existing.(armcompute.VirtualMachine); !ok {
+			return nil, errors.Errorf("%T is not an armcompute.VirtualMachine", existing)
 		}
 		// vm already exists
 		return nil, nil
```

Comment:
This is my best guess as to whats causing the controller error logs. `CreateOrUpdateAsync` in the client expects this return value to be `armcompute.VirtualMachine` and returns the logged error if not. I think that check in `CreateOrUpdateAsync` wasn't getting reached before because we used to only call that when the operation starts, but now we call it each loop to check if the operation is done.